include/backend/drm/drm.h

```diff
@@ -96,6 +97,7 @@ enum wlr_drm_connector_state {
 	WLR_DRM_CONN_NEEDS_MODESET,
 	WLR_DRM_CONN_CLEANUP,
 	WLR_DRM_CONN_CONNECTED,
+	WLR_DRM_CONN_LEASED
```

Comment:
Not sure we need a new state here. Maybe `DISCONNECTED` is enough? (We can always check `conn->lease_id`)